### ISSUE TITLE
Fun and Games with Minecraft on ARM

### ISSUE THREAD

**Issue by moocow1452** [lunixbochs/glshim#38]
So I've been trying for a while yet to run a full PC version of Minecraft on a SoC board like the Beaglebone Black or the Raspberry Pi, with the intention of making it into a portable build, so GLShim is an awfully useful project in that regard. I've seen video of it running on an Open Pandora, so I know that certain builds of certain versions of Minecraft can run, but can this functionality be ported to a more generic ARM board, or even allow for GLShim equipped application-ports for Android? 

PS: Most recent run on a LWJGL-patched Pi. Same or similar error message ran when I tried to run it on my Beaglebone Black, so that's promising.

[20:11:03 INFO]: Client> [20:11:03] [Client thread/INFO]: LWJGL Version: 2.9.1
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> # A fatal error has been detected by the Java Runtime Environment:
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> #  SIGSEGV (0xb) at pc=0x00000000, pid=2488, tid=3057566832
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> # JRE version: Java(TM) SE Runtime Environment (7.0_60-b19) (build 1.7.0_60-b19)
[20:11:04 INFO]: Client> # Java VM: Java HotSpot(TM) Client VM (24.60-b09 mixed mode linux-arm )
[20:11:04 INFO]: Client> # Problematic frame:
[20:11:04 INFO]: Client> # C  0x00000000
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> # Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> # An error report file with more information is saved as:
[20:11:04 INFO]: Client> # /home/pi/.minecraft/hs_err_pid2488.log
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> # If you would like to submit a bug report, please visit:
[20:11:04 INFO]: Client> #   http://bugreport.sun.com/bugreport/crash.jsp
[20:11:04 INFO]: Client> # The crash happened outside the Java Virtual Machine in native code.
[20:11:04 INFO]: Client> # See problematic frame for where to report the bug.
[20:11:04 INFO]: Client> #
[20:11:04 INFO]: Client> glXGetProcAddress: glXGetCurrentContext not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXQueryExtension not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXQueryVersion not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXGetClientString not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXGetFBConfigs not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXChooseFBConfig not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXGetFBConfigAttrib not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXGetVisualFromFBConfig not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXCreateWindow not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXCreateContextAttribsARB not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXEnumerateVideoDevicesNV not found.
[20:11:04 INFO]: Client> glXGetProcAddress: glXBindVideoCaptureDeviceNV not found.
[20:11:04 ERROR]: Game ended with bad state (exit code 134)
[20:11:04 INFO]: Ignoring visibility rule and showing launcher due to a game crash
[20:11:04 INFO]: Deleting /home/pi/.minecraft/versions/1.7.10/1.7.10-natives-570003515733

**Comment by kika123** [comment]
try the full-blown Mesa for VC4 driver.

**Comment by Epica-Lolz** [comment]
Well, here's the issue with that: The Mesa driver is OpenGL, and typically, most arm machines/boards come with OpenGLES, which is a special version of OpenGL designed for phones/tablets. To compensate for that, they made glshim, which is something that translates OpenGLES to regular OpenGL (Albeit, an older version of OpenGL.)

**Comment by waddlesplash** [comment]
@moocow1452: Zhuowei Zhang has already done this, it's called Boardwalk, and you can find it in the Play store.

**Comment by lunixbochs** [comment]
Boardwalk uses glshim. This is a thread about the Raspberry Pi, not Android. There's also a working Pandora version of Minecraft.

**Comment by lunixbochs** [comment]
You should try again on the latest unstable glshim (which has much better Pi support).
